### PR TITLE
IndirectDataReduction - EnergyTransfer Post Plot and Save

### LINKED ISSUE #17255: IndirectEnergyTransfer - Post Execution Plot and Save
Currently the Indirect ISISEnergyTransfer interface has checkboxes for saving multiple file types. This can be changed to a drop down menu and a save push button so saving can be completed post algorithm execution.
The plotting menu can also be accompanied by a push button for post algorithm execution and the "none" option removed.
These buttons and menus can be initially disabled and only enabled once the algorithm has completed.
Currently the plotting and saving is executed in the algorithm; this can be refactored to the interface.

### PR DESCRIPTION
**Pull request by louisemccann**
The plotting and saving of the results workspace has been changed to after the execution of the algorithm for the Indirect Energy Transfer tab.

**To test:**

Runs: 26176, 26174, 26173
\olympic\babylon5\Public\Louise McCann\Energy transfer
Run algorithm and check plot results as expected.
Also check all save formats are saved appropriately. 

Fixes #17255.

Release notes will be updated once post plot and save has been completed for all Indirect tabs.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by gvardany** [comment]
I am happy to review, but can you please make the test data available to non-ISIS developers?

**Comment by louisemccann** [comment]
@gvardany  \\olympic\babylon5\Public\Louise McCann\Energy transfer

**Comment by gvardany** [comment]
I meant for the access outside ISIS (e.g. ILL), will that be possible?

**Comment by louisemccann** [comment]
If you go to your build folder; Build\ExternalData\Testing\Data\UnitTest there should be IRS26173.raw and IRS26176.raw that you can use, hope this helps

File: MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Indirect/ISISEnergyTransfer.h [hunk 1]
```diff
@@ -70,6 +73,8 @@ private slots:
       const QString &
           groupType); ///< create the mapping file with which to group results
   std::vector<std::string> getSaveFormats(); ///< get a vector of save formats
+  std::vector<std::string>
+      outputWorkspaces; ///< get a vector of workspaces to plot
```
Comment (gvardany) [inline MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Indirect/ISISEnergyTransfer.h]: Shouldn't here be `m_outputWorkspaces`?

File: MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Indirect/ISISEnergyTransfer.ui [hunk 2]
```diff
@@ -819,6 +819,9 @@
         </item>
         <item>
          <widget class="QCheckBox" name="ckSaveSPE">
+          <property name="enabled">
+           <bool>false</bool>
+          </property>
           <property name="toolTip">
```
Comment (gvardany) [inline MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Indirect/ISISEnergyTransfer.ui]: Seems some of the checkboxes have tooltips, others don't. Can you make this consistent throughout?

File: MantidQt/CustomInterfaces/src/Indirect/ISISEnergyTransfer.cpp [hunk 3]
```diff
@@ -676,6 +687,45 @@ void ISISEnergyTransfer::pbRunFinished() {
 
   m_uiForm.dsRunFiles->setEnabled(true);
 }
+/**
+ * Handle mantid plotting of workspaces
+ */
+void ISISEnergyTransfer::plotClicked() {
+  for (auto it = outputWorkspaces.begin(); it != outputWorkspaces.end(); ++it) {
```
Comment (gvardany) [inline MantidQt/CustomInterfaces/src/Indirect/ISISEnergyTransfer.cpp]: Loops hereafter, why not `const auto & it : container`?

**Comment by gvardany** [comment]
I tested with the 2 files, seems to work as expected. See few minor comments inline.

**Comment by gvardany** [comment]
@louisemccann changes are better, but in the loops I think `const auto &` is more correct than looping by value.

**Comment by gvardany** [comment]
Changes seem fine, happy to :shipit:

**Pull request merged**